webpack/assets/javascripts/react_app/components/TemplateGenerator/TemplateGeneratorReducer.js

```diff
@@ -0,0 +1,41 @@
+import Immutable from 'seamless-immutable';
+
+import {
+  TEMPLATE_GENERATE_REQUEST,
+  TEMPLATE_GENERATE_POLLING,
+  TEMPLATE_GENERATE_SUCCESS,
+  TEMPLATE_GENERATE_FAILURE,
+} from './TemplateGeneratorConstants';
+
+const initialState = Immutable({
+  scheduleInProgress: false,
+  polling: false,
+  dataUrl: null,
+});
+
+export default (state = initialState, { type, payload }) => {
+  switch (type) {
+    case TEMPLATE_GENERATE_REQUEST:
+      return state.merge({ scheduleInProgress: true });
```

Comment:
it's only one value, how about using `set` here?`

Reply:
:dart: thanks